### PR TITLE
iamPolicyVersion needed to be configurable

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/6927

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Add iam_policy_version to iam_policy to make iamPolicyVersion configurable
```

### CODE REVIEW

File: api/resource/iam_policy.rb [hunk 1]
```diff
@@ -90,6 +90,9 @@ class IamPolicy < Api::Object
       # variables that are outside of the base_url qualifiers.
       attr_reader :import_format
 
+      # [Optional] Version number in the request payload.
+      # if set, it override the default iamPolicyVersion
```
Comment (danawillow) [inline api/resource/iam_policy.rb:94]:
```suggestion
      # if set, it overrides the default iamPolicyVersion
```

File: templates/terraform/iam_policy.go.erb [hunk 2]
```diff
@@ -178,14 +178,14 @@ func (u *<%= resource_name -%>IamUpdater) GetResourceIamPolicy() (*cloudresource
...
-	url, err = addQueryParams(url, map[string]string{"optionsRequestedPolicyVersion": fmt.Sprintf("%d", iamPolicyVersion)})
+       url, err = addQueryParams(url, map[string]string{"optionsRequestedPolicyVersion": fmt.Sprintf("%d", <% if object.iam_policy.iam_policy_version -%> <%= object.iam_policy.iam_policy_version -%> <% else -%> iamPolicyVersion <% end -%>)})
 	if err != nil {
 		return nil, err
 	}
 <% elsif object.iam_policy.iam_conditions_request_type == :REQUEST_BODY -%>
 	obj = map[string]interface{}{
 		"options": map[string]interface{}{
-			"requestedPolicyVersion": iamPolicyVersion,
+                       "requestedPolicyVersion": <% if object.iam_policy.iam_policy_version -%> <%= object.iam_policy.iam_policy_version -%> <% else -%> iamPolicyVersion <% end -%>,
```
Comment (danawillow) [inline templates/terraform/iam_policy.go.erb:188]: nit: it looks like some tabs got replaced with spaces. our code style is to use tabs for go and spaces for ruby, so please put the tabs back (and line 181)

File: templates/terraform/iam_policy.go.erb [hunk 3]
```diff
@@ -210,6 +210,10 @@ func (u *<%= resource_name -%>IamUpdater) SetResourceIamPolicy(policy *cloudreso
 		return err
 	}
 
+<% if object.iam_policy.iam_policy_version -%>
+        json["version"] = <%= object.iam_policy.iam_policy_version -%>
```
Comment (danawillow) [inline templates/terraform/iam_policy.go.erb:214]: If we didn't need this here before, why do we now?

File: templates/terraform/iam_policy.go.erb (hunk 2 above)
Comment (edwardmedia) [reply templates/terraform/iam_policy.go.erb]: yeah. I am fixing these now. Hard to see in my ide

File: templates/terraform/iam_policy.go.erb (hunk 3 above)
Comment (edwardmedia) [reply templates/terraform/iam_policy.go.erb:214]: This is for setIamPolicy(). The version is set outside of this file. Better to override it here for individual resource. Above two changes are for getIamPolicy().

Comment (danawillow) [reply templates/terraform/iam_policy.go.erb:214]: If it's being set outside of this file, then we should update the place where it _is_ set.

Comment (edwardmedia) [reply templates/terraform/iam_policy.go.erb:214]: good point!. let me check

Comment (danawillow) [reply templates/terraform/iam_policy.go.erb:214]: can you add a comment explaining that this is an override of the existing version that might have been set in the resource_iam_member|policy|binding code?

**Pull request merged**